### ISSUE TITLE
mxnet zoo module is failing very hard because of dependecies (unsure which api ver)

### ISSUE THREAD

**Issue by LJWxyz** [awslabs/djl#122]
Dependencies hell by>MxModelZoo or ai.djl.repository.RepositoryFactoryImpl

**D:\Coding\JavaSourceCode\user\plugin>java SimplelocalSSDeg
D:/Coding/JavaSourceCode/user/plugin/djlssdin\in.png
Exception in thread "main" java.lang.NoClassDefFoundError: org/slf4j/LoggerFactory
        at ai.djl.repository.RepositoryFactoryImpl.<clinit>(RepositoryFactoryImpl.java:32)
        at ai.djl.repository.Repository.newInstance(Repository.java:96)
        at ai.djl.mxnet.zoo.MxModelZoo.<clinit>(MxModelZoo.java:50)
        at SimplelocalSSDeg.objdtctwlegacyssdmodel(SimplelocalSSDeg.java:20)
        at SimplelocalSSDeg.main(SimplelocalSSDeg.java:16)
Caused by: java.lang.ClassNotFoundException: org.slf4j.LoggerFactory
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:602)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
        ... 5 more

D:\Coding\JavaSourceCode\user\plugin>cd D:\Coding\JavaRuntimeAsset\djl0_60\otherpackagedependencies

D:\Coding\JavaRuntimeAsset\djl0_60\otherpackagedependencies>set CLASSPATH=%CLASSPATH%;%cd%\*

D:\Coding\JavaRuntimeAsset\djl0_60\otherpackagedependencies>cd D:\Coding\JavaSourceCode\user\plugin

D:\Coding\JavaSourceCode\user\plugin>java SimplelocalSSDeg
D:/Coding/JavaSourceCode/user/plugin/djlssdin\in.png
Exception in thread "main" java.lang.NoClassDefFoundError: org/slf4j/LoggerFactory
        at ai.djl.repository.RepositoryFactoryImpl.<clinit>(RepositoryFactoryImpl.java:32)
        at ai.djl.repository.Repository.newInstance(Repository.java:96)
        at ai.djl.mxnet.zoo.MxModelZoo.<clinit>(MxModelZoo.java:50)
        at SimplelocalSSDeg.objdtctwlegacyssdmodel(SimplelocalSSDeg.java:20)
        at SimplelocalSSDeg.main(SimplelocalSSDeg.java:16)
Caused by: java.lang.ClassNotFoundException: org.slf4j.LoggerFactory
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:602)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
        ... 5 more

D:\Coding\JavaSourceCode\user\plugin>java SimplelocalSSDeg
D:/Coding/JavaSourceCode/user/plugin/djlssdin\in.png
Exception in thread "main" java.util.ServiceConfigurationError: org.slf4j.spi.SLF4JServiceProvider: Provider org.slf4j.log4j12.Log4j12ServiceProvider could not be instantiated
        at java.base/java.util.ServiceLoader.fail(ServiceLoader.java:584)
        at java.base/java.util.ServiceLoader$ProviderImpl.newInstance(ServiceLoader.java:806)
        at java.base/java.util.ServiceLoader$ProviderImpl.get(ServiceLoader.java:724)
        at java.base/java.util.ServiceLoader$3.next(ServiceLoader.java:1396)
        at org.slf4j.LoggerFactory.findServiceProviders(LoggerFactory.java:104)
        at org.slf4j.LoggerFactory.bind(LoggerFactory.java:147)
        at org.slf4j.LoggerFactory.performInitialization(LoggerFactory.java:139)
        at org.slf4j.LoggerFactory.getProvider(LoggerFactory.java:418)
        at org.slf4j.LoggerFactory.getILoggerFactory(LoggerFactory.java:404)
        at org.slf4j.LoggerFactory.getLogger(LoggerFactory.java:353)
        at org.slf4j.LoggerFactory.getLogger(LoggerFactory.java:379)
        at ai.djl.repository.RepositoryFactoryImpl.<clinit>(RepositoryFactoryImpl.java:32)
        at ai.djl.repository.Repository.newInstance(Repository.java:96)
        at ai.djl.mxnet.zoo.MxModelZoo.<clinit>(MxModelZoo.java:50)
        at SimplelocalSSDeg.objdtctwlegacyssdmodel(SimplelocalSSDeg.java:20)
        at SimplelocalSSDeg.main(SimplelocalSSDeg.java:16)
Caused by: java.lang.NoClassDefFoundError: org/apache/log4j/Level
        at org.slf4j.log4j12.Log4j12ServiceProvider.<init>(Log4j12ServiceProvider.java:27)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
        at java.base/java.util.ServiceLoader$ProviderImpl.newInstance(ServiceLoader.java:782)
        ... 14 more
Caused by: java.lang.ClassNotFoundException: org.apache.log4j.Level
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:602)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
        ... 21 more

D:\Coding\JavaSourceCode\user\plugin>**

## How to Reproduce?
ZooModel<ai.djl.modality.cv.Image,ai.djl.modality.cv.output.DetectedObjects> model = MxModelZoo.SSD.loadModel(new ProgressBar());//Get resnet model from model zoo.
^This MxModelZoo is refering to (ai.djl.mxnet.zoo.MxModelZoo)
It's either the 0.60core api/api0.21/or any 0.60 submodule

### Steps to reproduce
get the some class to run ^code,
clean all your system&env var to not include any jar except those i mentioned
add (openjdkzulu14\bin)>classpath
then follow my terminal flow
(you should also face gson dependencies error but it's easy to fix just download the latest ver jar add it to classpath,solved)d:07072020
Complete command in terminal can be observed at the stacktrace^ no filter of my all attempts to fix

## What have you tried to solve it?
in summary (downloading the required dependencies put to classpath retry call class>stacktraceout)repeat

openjdkzulu14\bin;
(all jar in D:\Coding\JavaRuntimeAsset\djl0_60 which have)>
api-0.2.1,DJL-0.6.0-SSHOT,examples-0.6.0,model-zoo-0.6.0,mxnet-engine-0.6.0,mxnet-model-zoo-0.6.0,pytorch-engine-0.6.0,pytorch-model-zoo-0.6.0,(and those tensorflow jar)

Suggestion>break off from those inconsistent logging api?At release code(jar)/(simply production code)
Especially (org/slf4j)series,they have a very complicated release system that normally can destroy any newcomer that are interested but when stacktrace come confirm,Instant KO.

**Comment by frankfliu** [comment]
This exception is not caused by DJL.

DJL doesn't directly depends on log4j, DJL depends on slf4j-api: https://mvnrepository.com/artifact/org.slf4j/slf4j-api/1.7.30

It's developer's responsibility to choose which log binding to use.
The popular choice are:
- logback: https://mvnrepository.com/artifact/ch.qos.logback/logback-core/1.2.3
- log4j2: https://mvnrepository.com/artifact/org.apache.logging.log4j/log4j-slf4j-impl/2.13.3

I noticed your stacktrace shows you are using log4j1.2, which is deprecated already. I think you included a mismatched log4j version and slf4j-log4j binding in your classpath. This is common mistake the using log4j. But we don't have control of this in DJL.

**Comment by LJWxyz** [comment]
Thanks for the helpful reply.It helped me to get it progress!oh yeah.
so this issue can now mainly be a reminder to update documentation to guide newcomer about the extra required dependencies.
In my folder now have>common compress-1.2,gson-2.8.6,slf4j-api-1.7.30.
Up until now i had never used any logging tools.ahha .I only uses exception.I do however see the point of using logging now.but i personally will just stick with jdk stuff to remain consistent and elegant.ahhaa.
Once again,once the documentation is done this issue finish serving it's purpose and get closed.^^

**Comment by frankfliu** [comment]
I created a README about log4j issue: https://github.com/awslabs/djl/blob/master/docs/development/configure_logging.md#log4j-version-clash

Feel free to reopen this issue if it's not clear to you.
